src/app/views/settings/apps/apps.controller.coffee

```diff
@@ -3,9 +3,35 @@
   vm = this
 
   vm.enabledApps = []
+  vm.filteredApps = []
+  vm.selectedCategory = ''
+  vm.searchTerm = ''
 
   vm.tenantManagement = false
 
+  # Filter products by name or category
+  vm.onSearchChange = () ->
+    vm.selectedCategory = ''
+    vm.filteredApps = []
+    firstFilterResult = []
+    for app in vm.enabledApps
+      if (vm.searchTerm? && vm.searchTerm.length > 0) || !vm.selectedCategory
```

Comment:
You're doing this test on every iteration of the `for` loop but the result is not dependent on the `app`